### DISCUSSION TITLE
请问该怎么解决 CODE: 2  DESC: [REJECTREQUEST]system busy, start flow control for a while BROKER:

### DISCUSSION THREAD

**Discussion by yijieyu** [apache/rocketmq#6656] [Q&A]
broker_default.log 一直报system busy
服务器2核8G
版本5.1.0
用controller方式quick_start启动

这是配置文件

brokerClusterName = Cluster
brokerName = rocket01
brokerId = -1
brokerRole = ASYNC_MASTER
#flushDiskType=ASYNC_FLUSH
deleteWhen = 04
fileReservedTime = 48
enableControllerMode = true
controllerAddr = 10.203.1.115:9878;10.203.0.175:9878;10.203.1.50:9878
namesrvAddr = 10.203.1.115:9876;10.203.0.175:9876;10.203.1.50:9876
allAckInSyncStateSet=true
listenPort=30911
storePathRootDir=/data/rmqstore/rocket02/
storePathCommitLog=/data/rmqstore/rocket02/commitlog
sendMessageThreadPoolNums=4
useReentrantLockWhenPutMessage=true
isEnableSlaveActingMaster=true

autoCreateTopicEnable=false
autoCreateSubscriptionGroup=false

**Comment by imp2002** [comment]
![image](https://user-images.githubusercontent.com/41513919/234561217-d71ef767-a460-43d5-ac59-9855c216a564.png)

Seems cause by `Page Cache busy`, a possible solution is to enable the `transientStorePoolEnable` and set the `transientStorePoolSize` realize read-write separation, relieve pressure on page cache.
```conf
transientStorePoolEnable=true
transientStorePoolSize=1
```

**Reply by yijieyu** [reply]
okkk

**Reply by ShadowySpirits** [reply]
Thanks @imp2002, Great suggestion!

But I would like to recommend setting the `transientStorePoolSize` >= 2 to make sure every CommitLog could borrow a write buffer.

And maybe you want to know the trade-off for enabling transientStorePool: each buffer will occupy 1G memory by default, and increase the latency of flushing disk (setting flushDiskType=SYNC_FLUSH could be better).

**Comment by RongtongJin** [comment]
> broker_default.log 一直报system busy 服务器2核8G 版本5.1.0 用controller方式quick_start启动
> 
> 这是配置文件
> 
> brokerClusterName = Cluster brokerName = rocket01 brokerId = -1 brokerRole = ASYNC_MASTER #flushDiskType=ASYNC_FLUSH deleteWhen = 04 fileReservedTime = 48 enableControllerMode = true controllerAddr = 10.203.1.115:9878;10.203.0.175:9878;10.203.1.50:9878 namesrvAddr = 10.203.1.115:9876;10.203.0.175:9876;10.203.1.50:9876 allAckInSyncStateSet=true listenPort=30911 storePathRootDir=/data/rmqstore/rocket02/ storePathCommitLog=/data/rmqstore/rocket02/commitlog sendMessageThreadPoolNums=4 useReentrantLockWhenPutMessage=true isEnableSlaveActingMaster=true
> 
> autoCreateTopicEnable=false autoCreateSubscriptionGroup=false

2c8g在较大流量下可能确实会造成system busy。调大osPageCacheBusyTimeOutMills可能可以从表面上缓解该现象，但如果系统存在性能瓶颈，还是需要升级规格或扩容。此外，我发现配置文件中isEnableSlaveActingMaster参数和enableControllerMode同时为true，两种模式无法同时启动，建议关闭isEnableSlaveActingMaster参数。

**Reply by yijieyu** [reply]
我试试
